src/libcore/macros.rs

```diff
@@ -94,7 +94,28 @@ macro_rules! assert_eq {
                 }
             }
         }
-    })
+    });
+    ($left:expr , $right:expr, $msg:expr) => ({
+        match (&($left), &($right)) {
+            (left_val, right_val) => {
+                if !(*left_val == *right_val) {
+                    panic!("assertion failed: `(left == right)` \
+                           (left: `{:?}`, right: `{:?}`): {}", left_val, right_val, $msg)
```

Comment:
It may be worth using `concat!` here as well to ensure that a `$msg` which looks like `"foo {}"` gives an error saying that it expected an argument rather than being accepted